python/tvm/relay/quantize/_annotate.py

```diff
@@ -156,6 +156,18 @@ def conv2d_rewrite(ref_call, new_args, ctx):
     if cnt < current_qconfig().skip_k_conv:
         _set_conv_counter(cnt + 1)
         return None
+
+    if current_qconfig().skip_conv_layers is not None:
+        leave_alone_indices = [int(x) for x in current_qconfig().skip_conv_layers]
+        if cnt in leave_alone_indices:
+            _set_conv_counter(cnt + 1)
+            return None
+
+    # Check if the kernel is suitable for quantization, post warning if not.
+    in_channels = new_args[1].data.shape[1]
+    if in_channels % 4 != 0:
```

Comment:
Can you explain why in channels divisible by 4 is required?

Reply:
if input channels isn't divisible by 4 relay fails to convert from NCHW to NCHWc and throws an error. This is ok if not using dp4a but annoying otherwise. Printing out a simple warning and count in these cases makes it easier to use the skip_conv_layers option if needed and is easy to ignore if not.

Reply:
I would suggest removing the warning here since it is unrelated to quantization. The default conv2d implementation conv2d can work with different types. So if input channels isn't divisible by 4 (or other factor) we should use the NCHW one. We have some assertions like [this](https://github.com/dmlc/tvm/blob/134a2f25d6305036cd2bd7d63caa555c7dbcbabf/topi/python/topi/cuda/conv2d_int8.py#L73) in NCHWc conv2d. However, we don't expect that any errors will be raised since you have no way to obtain the AutoTVM log using invalid true. So NCHWc one shouldn't be chosen in this case.

Reply:
That's fair, I've reverted the warning.